### PR TITLE
zb: replace map with sync.Map to avoid concurrent writes closes #582

### LINKED ISSUE #582: zb: concu
**Describe the bug**
Race detected in zb: concurrent map writes

**Additional context**

https://github.com/project-zot/zot/runs/6758810995?check_suite_focus=true#step:13:208

```
fatal error: concurrent map writes
goroutine 635 [running]:
runtime.throw({0x563ac0411dda, 0x563abfccd874})
	runtime/panic.go:1198 +0x71 fp=0xc0002cf848 sp=0xc0002cf818 pc=0x563abfbdee31
runtime.mapassign_faststr(0xc0003b1578, 0x3, {0x563ac03fc50c, 0x4})
	runtime/map_faststr.go:[211](https://github.com/project-zot/zot/runs/6758810995?check_suite_focus=true#step:13:212) +0x39c fp=0xc0002cf8b0 sp=0xc0002cf848 pc=0x563abfbbc0dc
main.pushMonolithAndCollect.func1(0xc000127b80, {0x0, 0x30}, 0xc00059cdf8, 0xc00018c000, {0x7ffe73ffc023, 0x15}, {{0x563ac0430cd2, 0x30}, 0x563ac0c08398, ...}, ...)
	zotregistry.io/zot/cmd/zb/helper.go:475 +0x69e fp=0xc0002cfd58 sp=0xc0002cf8b0 pc=0x563ac03f1ebe
main.pushMonolithAndCollect({0x0, 0x563ac0b09640}, {0x7ffe73ffc023, 0xc002d3a9c0}, {0x0, 0xc0001c4500}, 0x563ac0ab14c0, {0xc0001c4520, 0x1, 0x1}, ...)
	zotregistry.io/zot/cmd/zb/helper.go:617 +0xa5 fp=0xc0002cfdf8 sp=0xc0002cfd58 pc=0x563ac03f1765
main.PushMonolithStreamed({0x0, 0x0}, {0x7ffe73ffc023, 0x15}, {0x0, 0x0}, 0x5, {{0x563ac0430cd2, 0x30}, 0x563ac0c08398, ...}, ...)
	zotregistry.io/zot/cmd/zb/perf.go:365 +0x14e fp=0xc0002cfec0 sp=0xc0002cfdf8 pc=0x563ac03f818e
main.Perf.func1()
	zotregistry.io/zot/cmd/zb/perf.go:686 +0x20d fp=0xc0002cffe0 sp=0xc0002cfec0 pc=0x563ac03f9d2d
runtime.goexit()
	runtime/asm_amd64.s:1581 +0x1 fp=0xc0002cffe8 sp=0xc0002cffe0 pc=0x563abfc0f761
created by main.Perf
	zotregistry.io/zot/cmd/zb/perf.go:678 +0x8bc
```

### PR DESCRIPTION
**Pull request by peusebiu**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/zb/perf.go [hunk 1]
```diff
@@ -48,7 +48,26 @@ const (
 var blobHash map[string]godigest.Digest = map[string]godigest.Digest{}
 
 // nolint:gochecknoglobals // used only in this test
-var statusRequests map[string]int
+var statusRequests statusRequestsMap
```
Comment (rchincha) [inline cmd/zb/perf.go:51]: Why not just declare this to be a sync.Map instead of making a struct/interface

File: cmd/zb/perf.go [hunk 2]
```diff
@@ -48,7 +48,18 @@ const (
 var blobHash map[string]godigest.Digest = map[string]godigest.Digest{}
 
 // nolint:gochecknoglobals // used only in this test
-var statusRequests map[string]int
+var statusRequests sync.Map
+
+func loadOrStore(statusRequests *sync.Map, key string, value int) int {
```
Comment (rchincha) [inline cmd/zb/perf.go:53]: move this into the helper.go as well

Comment (rchincha) [reply cmd/zb/perf.go:53]: try to keep perf.go clean

Comment (peusebiu) [reply cmd/zb/perf.go]: fixed, thank you.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**